src/xdocs/config_metrics.xml

```diff
@@ -696,7 +696,7 @@ class Foo {
             <tr>
               <td>excludeClassesRegexps</td>
               <td>Specify user-configured regular expressions to ignore classes.</td>
-              <td><a href="property_types.html#regexp">Regular Expressions</a></td>
+              <td><a href="property_types.html#Pattern.5B.5D">Pattern[]</a></td>
```

Comment:
just wonder - shouldn't this be `%5B%5D` instead?

Reply:
https://github.com/checkstyle/checkstyle/blob/cf08aa01021f064a99e7816a280ba89143623442/src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java#L152

it is `String[]` in setter, so  real type is `String[]`, if we want  to mean regexp nature of it keep it `Regular Expressions`.
Please do the same by all others.

PLEASE keep in this PR only updates where setter type is `Pattern`, all other cases keep as  `Regular Expression` or ` `Regular Expressions`.

Reply:
@romani 
That is what I tried to mention here in my comment https://github.com/checkstyle/checkstyle/issues/8328#issuecomment-652964231 that there are more than 5 such cases where setter type and field class is different.

Reply:
lets skip all complicated cases for now, and merge what is simple.
in this PR please skip all such cases, ones telegram discussion finished, we will tackle them.

Reply:
Done

Reply:
@gaurabdg , please answer question from @strkkk .

Reply:
> just wonder - shouldn't this be `%5B%5D` instead?

The generated site contains link in this format.